File: backend/app/schemas/voice.py
from __future__ import annotations

from app.schemas.base import CamelModel
from app.schemas.persona import Persona  # noqa: TC001 - Pydantic resolves field types at runtime
from app.schemas.product import Recommendation  # noqa: TC001 - Pydantic resolves field types at runtime


class VoiceTokenResponse(CamelModel):
    """Ephemeral token for Gemini Live API WebSocket connection."""

    token: str
    model: str


class TranscriptMessage(CamelModel):
    """A single message from a voice conversation transcript."""

    role: str
    content: str


class TranscriptRequest(CamelModel):
    """Request body for voice transcript processing."""

    session_id: str
    messages: list[TranscriptMessage]


class TranscriptResponse(CamelModel):
    """Result of processing a voice transcript through the persona pipeline."""

    session_id: str
    persona: Persona | None = None
    recommendations: list[Recommendation] = []

File: frontend/src/types/voice.ts
import { z } from "zod";
import { PersonaSchema } from "./persona";
import { RecommendationSchema } from "./product";

export const VoiceTokenResponseSchema = z.object({
  token: z.string(),
  model: z.string(),
});

export type VoiceTokenResponse = z.infer<typeof VoiceTokenResponseSchema>;

export const TranscriptMessageSchema = z.object({
  role: z.enum(["user", "assistant"]),
  content: z.string(),
});

export type TranscriptMessage = z.infer<typeof TranscriptMessageSchema>;

export const TranscriptResponseSchema = z.object({
  sessionId: z.string(),
  persona: PersonaSchema.nullable().default(null),
  recommendations: z.array(RecommendationSchema).default([]),
});

export type TranscriptResponse = z.infer<typeof TranscriptResponseSchema>;
